portfolio/src/main/webapp/script.js

```diff
@@ -16,120 +16,195 @@
-            commentsListElement.appendChild(createCommentElement(comment));
-        })
-    });
+  fetch('/data').then(response => response.json()).then((commentsArray) => {
+    const commentsListElement = document.getElementById('comments-list');
+    currentComments = commentsArray;
+    commentsArray.forEach((comment) => {
+      commentsListElement.appendChild(createCommentElement(comment));
+    })
+  });
 }
 
+// Creating comment element with delete button 
 function createCommentElement(comment) {
-    const commentElement = document.createElement('li');
-    commentElement.className = 'comment';
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const usernameElement = document.createElement('div');
+  usernameElement.innerText = comment.username;
+  usernameElement.className = 'namediv';
+  const messageElement = document.createElement('div');
+  messageElement.innerText = comment.message;
+  messageElement.className = 'messagediv';
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+  let test = prompt("Enter passcode:", "");
+    if(test == "1234"){
+      deleteComment(comment);
+      var index = currentComments.indexOf(comment);
+      currentComments.splice(index,1);
+      commentElement.remove();
+    }
+  });
+
+  commentElement.appendChild(usernameElement);
+  usernameElement.appendChild(deleteButtonElement);
+  commentElement.appendChild(messageElement);
 
-    const usernameElement = document.createElement('span');
-    usernameElement.innerText = comment.username;
-    usernameElement.className = 'namespan';
-    const messageElement = document.createElement('span');
-    messageElement.innerText = comment.message;
-    usernameElement.className = 'messagespan';
+  return commentElement;
+}
 
-    commentElement.appendChild(messageElement);
-    commentElement.appendChild(usernameElement);
+// Deleting a comment from server
+function deleteComment(comment) {
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-comment', {method: 'POST', body: params});
+}
+
+function getData() {
+  var xhr = new XMLHttpRequest();                 // Create XMLHttpRequest object
+
+  xhr.onload = function() {                       // When response has loaded
+    // The following conditional check will not work locally - only on a server
+    if(xhr.status === 200) {
+      var text = xhr.responseText;                   // If server status was ok
+      refreshComments(JSON.parse(text)); // Update
+    }
+  };
+
+  xhr.open('GET', '/data', true);        // Prepare the request
+  xhr.send(null);                                 // Send the request
+}
+
+// Refreshes the data each 2 seconds
+setInterval(getData, 1000*2);
+
+// New comments(=comments) are added to the page, and old comments(=currentComments) are removed
```

Comment:
This algorithm is OK for a small number of comments, but keep in mind that it is O(N^2) in the number of comments. Please add a note about it, so, if performance becomes critical, we can move to a better algorithm. For example, using a dictionary for testing the objNotInList or having the comments always in a canonical representation (for example, sorted by date) which allows a O(N) algorithm for  comparing new and old version.

Reply:
I know how to do it O(n) on paper, but I got stuck in js. All possible deleted comments should be before the date of the youngest comment in the localComments(checking can be done with two pointers) . And the new ones after that value. This might not work.